### ISSUE TITLE
::tarray::table delete causes a column shift

### ISSUE THREAD

**Issue by LukasHoerger** [apnadkarni/tarray#1]
Hello Ashok,

thank you for this great package! I stumbled upon a bug with the table delete functions.
If you delete the first row of a table with at least one boolean column the columns of the last table row will shift, omitting the values of the boolean column.

## Steps to Reproduce
Consider the simple table `table` with two columns, the first being a boolean column.
Deleting the row with index `0` causes the column shift. Deleting with low and high indexes, eg. `0` and `1` also results in the same problem. If the row gets deleted with an index list, no column shift occurs. Following example demonstrates the problem:

```tcl
set table [::tarray::table create {
    A boolean B boolean C int
} {
	{0 1 10}
	{0 1 20}
	{0 1 30}
}]

# column shift
set table [::tarray::table delete $table 0]

# column shift
# set table [::tarray::table delete $table 0 1]

# no column shift
# set table [::tarray::table delete $table {0 1}]

set pretty [::tarray::table prettify $table]
```

## Expected Behavior
I expect the content of the variable `pretty` to look like this:
```
+--+-+--+
|A|B| C|
+--+-+--+
|0|1|20|
+--+-+--+
|0|1|30|
+--+-+--+
```
## Current Behavior
Currently `pretty` looks like this:
```
+--+-+--+
| A|B| C|
+--+-+--+
| 0|1|20|
+--+-+--+
|30| |  |
+--+-+--+
```

## Context (Environment)
I am running tcl 8.6.10 and the package version 1.0.0 on Windows 11.

My C-knowledge is not so great, but I suspect the problem could be in thdr_delete_range and/or thdr_delete_indices. If there is anything else I can help you with, feel free to let me know.

Best regards
Lukas

**Comment by apnadkarni** [comment]
![tarray](https://github.com/user-attachments/assets/25f0344d-2090-4425-8c23-c8c6e500ee88)

Running Tcl 8.6.14 on Windows 10, I don't see this. See the attached screen shot. Is it possible for you to double check with Tcl 8.6.14 though I doubt that would matter. Also, are you running 32- or 64-bit?

**Comment by apnadkarni** [comment]
Ok, I can reproduce it. But only if I run your sample as a script, not interactively. Also the bug seems to be in the prettify command, not the delete. Can you show me what `puts $table` prints?

**Comment by apnadkarni** [comment]
I have enough to go on debugging. Thanks for the report.

**Comment by LukasHoerger** [comment]
Thank you for your quick response. I don't have a 8.6.14 right now, but I tested it with 32 and 64 bit, it is the same. After your hint with the prettify, I looked into it and found, that `::tarray::table index $table 1` yields `0 1 30`, but `::tarray::table rows $table` yields `{0 1 20} 30` instead of `{0 1 20} {0 1 30}`. If I can be of further help, let me know! Thanks and have a great weekend!
